### PR TITLE
std.functional added missing compare functions

### PR DESCRIPTION
**Pull request by burner**
* lessOrEqualThan
* greaterOrEqualThan
* notEqualTo
* added unittest comments

### CODE REVIEW

**Comment by JackStouffer** [comment]
Add these new aliases in the top ddoc, then LGTM.

File: std/functional.d [hunk 1]
```diff
@@ -26,6 +26,9 @@ $(TR $(TH Function Name) $(TH Description)
     $(TR $(TD $(D $(LREF lessThan)), $(D $(LREF greaterThan)), $(D $(LREF equalTo)))
         $(TD Ready-made predicate functions to compare two values.
     ))
+    $(TR $(TD $(D $(LREF lessOrEqualThan)), $(D $(LREF greaterOrEqualThan)), $(D $(LREF notEqualTo)))
```
Comment (JackStouffer) [inline std/functional.d]: Why two different sections? They have the same descriptions, why not merge them?

Comment (burner) [inline std/functional.d]: very long table column, take a look at the doc generator

Comment (JackStouffer) [inline std/functional.d]: IMO it makes more sense to have a long column than to have two sections with identical descriptions.

Comment (burner) [inline std/functional.d]: I disagree, it will make the overall takes hard to read

Comment (CyberShadow) [inline std/functional.d]: @burner Does your browser not wrap the column?

JackStouffer is right, it looks like a lazy copy-paste job right now. There are better solutions for this problem.

**Comment by CyberShadow** [comment]
`lessOrEqualThan` - `not!greaterThan`
`greaterOrEqualThan` - `not!lessThan`
`notEqualTo` - `not!equalTo`

Is this really needed?

File: std/functional.d (hunk 1 above)
Comment (burner) [inline std/functional.d]: it does wrap the lines, I just copied the already present style and I think it looks good this way.

I'll change it anyway

**Comment by burner** [comment]
@CyberShadow I would say it is needed, otherwise you could ask to delete the other three as well. And, I actually needed the other three. Doing it the "not" route is just another mental step people (I) would have to do.

**Comment by burner** [comment]
updated to a single line

**Comment by JackStouffer** [comment]
LGTM

**Comment by andralex** [comment]
These seem just too trivial. Are we sure they're actually useful?

**Comment by burner** [comment]
If I'm a significant D user group than yes. I had lessThan in some of my code and needed lessOrEqualThan. I was disappointed to find it missing.

**Comment by quickfur** [comment]
I dunno... `safeOp!"<="` is easier to type than `lessThanOrEqual`, and more obvious exactly what it being done. I'm with @andralex that these aliases seem too trivial an addition.

(Not to mention, in English it should be `lessThanOrEqual`, not `lessOrEqualThan`. Ditto for the others.)

**Pull request closed without merging**